### PR TITLE
SDL: use screen size in fullscreen

### PR DESCRIPTION
**Pull request by vvanelslande**
Fixes this bug reported in [#citra-support](https://discordapp.com/channels/220740965957107713/220742362740162560/645645566327521313):
> instead of being fullscreen at 1366x768, it drops to fullscreen ~480x320 or 320x240

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/5003)

### CODE REVIEW

File: src/citra/emu_window/emu_window_sdl2.cpp [hunk 1]
```diff
@@ -138,12 +138,21 @@ EmuWindow_SDL2::EmuWindow_SDL2(bool fullscreen) {
...
-    render_window =
-        SDL_CreateWindow(window_title.c_str(),
-                         SDL_WINDOWPOS_UNDEFINED, // x position
-                         SDL_WINDOWPOS_UNDEFINED, // y position
-                         Core::kScreenTopWidth, Core::kScreenTopHeight + Core::kScreenBottomHeight,
-                         SDL_WINDOW_OPENGL | SDL_WINDOW_RESIZABLE | SDL_WINDOW_ALLOW_HIGHDPI);
+    int width, height;
+    if (fullscreen) {
+        SDL_DisplayMode display_mode;
+        SDL_GetCurrentDisplayMode(0, &display_mode);
```
Comment (wwylele) [inline src/citra/emu_window/emu_window_sdl2.cpp]: Do you happened to know how multiple monitors would interact with full screen? Is there a better solution than hardcoding "monitor 0" here?

Comment (vvanelslande) [reply src/citra/emu_window/emu_window_sdl2.cpp]: no. I have 1 monitor.

Comment (BreadFish64) [reply src/citra/emu_window/emu_window_sdl2.cpp]: you'll probably want to use [this](https://wiki.libsdl.org/SDL_GetWindowDisplayIndex) to figure out which display the window is on

Comment (adityaruplaha) [reply src/citra/emu_window/emu_window_sdl2.cpp]: Ah I wrote this code. Sorry about that.

Comment (vvanelslande) [reply src/citra/emu_window/emu_window_sdl2.cpp]: the `render_window` is created after getting the display mode

Comment (treymerkley) [reply src/citra/emu_window/emu_window_sdl2.cpp]: could this be done by creating a config variable that is whatever number the user sets it to, with a default of 0?

**Comment by vvanelslande** [comment]
`-x, --fullscreen-display-index     Default: 0` option added. uses 400x480 resolution if `SDL_GetCurrentDisplayMode(fullscreen_display_index, &display_mode) != 0`.

File: src/citra/emu_window/emu_window_sdl2.cpp [hunk 2]
```diff
@@ -245,7 +261,7 @@ void EmuWindow_SDL2::PollEvents() {
     }
 
     const u32 current_time = SDL_GetTicks();
-    if (current_time > last_time + 2000) {
+    if (!is_fullscreen && current_time > last_time + 2000) {
```
Comment (vvanelslande) [inline src/citra/emu_window/emu_window_sdl2.cpp]: fixes a crash

**Comment by vvanelslande** [comment]
I don't want to contribute to Citra anymore.

**Pull request closed without merging**